<launch>
    <!-- Detection configuration -->
    <arg name="weights" default="$(find yolov5_ros)/weights/1280_seg_m.pt" />
    <arg name="data" default="$(find yolov5_ros)/data/coco128.yaml"/>
    <arg name="confidence_threshold" default="0.5" />
    <arg name="iou_threshold" default="0.5" />
    <arg name="retina_masks" default="true" />
    <arg name="hide_labels" default="false" />
    <arg name="hide_conf" default="false" />
    <!-- <arg name="maximum_detections" default="1000" /> -->
    <arg name="device" default="0" />
    <arg name="agnostic_nms" default="true" />
    <arg name="line_thickness" default="3" />
    <arg name="dnn" default="true" />
    <arg name="half" default="false" />

    <!-- replace imgsz -->
    <arg name="inference_size_w" default="960"/>
    <arg name="inference_size_h" default="720"/>

    <!-- ROS input topics -->
    <!-- <arg name="input_image_topic" default="/camera/image_color/compressed"/> -->
    <arg name="input_image_topic" default="/camera/color/image_raw"/>
    
    <!-- choose the output type -->
    <arg name="if_publish_box" default="true"/>
    <arg name="if_publish_image" default="true"/>

    <!-- Optional topic (publishing annotated image) -->
    <arg name="output_image_topic" default="/yolov5/segment"/>
    <arg name="output_box_topic" default="/yolov5/predict/box"/>


    <node pkg="yolov5_ros" name="segment" type="segment_grey.py" output="screen">
        <param name="weights" value="$(arg weights)"/>
        <param name="data" value="$(arg data)"/>
        <param name="confidence_threshold" value="$(arg confidence_threshold)"/>
        <param name="iou_threshold" value="$(arg iou_threshold)" />
        <param name="retina_masks" value="$(arg retina_masks)" />
        <param name="hide_labels" value="$(arg hide_labels)" />
        <param name="hide_conf" value="$(arg hide_conf)" />
        <!-- <param name="maximum_detections" value="$(arg maximum_detections)"/> -->
        <param name="device" value="$(arg device)" />
        <param name="agnostic_nms" value="$(arg agnostic_nms)" />
        <param name="line_thickness" value="$(arg line_thickness)"/>
        <param name="dnn" value="$(arg dnn)"/>
        <param name="half" value="$(arg half)"/>

        <param name="inference_size_h" value="$(arg inference_size_h)"/>
        <param name="inference_size_w" value="$(arg inference_size_w)"/>

        <param name="input_image_topic" value="$(arg input_image_topic)"/>

        <param name="if_publish_box" value="$(arg if_publish_box)"/>
        <param name="if_publish_image" value="$(arg if_publish_image)"/>

        <param name="output_image_topic" value="$(arg output_image_topic)"/>
        <param name="output_box_topic" value="$(arg output_box_topic)"/>
    </node>
    <!-- <include file="$(find camera_launch)/launch/d435.launch"/> -->


</launch>
